ecosystem/typescript/sdk/packages/aptosClient/src/index.node.ts

```diff
@@ -0,0 +1,107 @@
+import { CookieJar } from "./cookieJar";
+import { AptosClientRequest, AptosClientResponse } from "./types";
+
+const cookieJar = new CookieJar();
+
+export default async function aptosClient<Res>(requestOptions: AptosClientRequest): Promise<AptosClientResponse<Res>> {
+  const { params, method, url, headers, body } = requestOptions;
+
+  const request: OptionsOfJSONResponseBody = {
+    http2: true,
+    searchParams: convertBigIntToNumber(params),
+    method,
+    url,
+    responseType: "json",
+    headers,
+    hooks: {
+      beforeRequest: [
+        (options) => {
+          const cookies = cookieJar.getCookies(new URL(options.url!));
+
+          if (cookies?.length > 0 && options.headers) {
+            /* eslint-disable no-param-reassign */
+            options.headers.cookie = cookies.map((cookie: any) => `${cookie.name}=${cookie.value}`).join("; ");
+          }
+        },
+      ],
+      afterResponse: [
+        (response) => {
+          if (Array.isArray(response.headers["set-cookie"])) {
+            response.headers["set-cookie"].forEach((c) => {
+              cookieJar.setCookie(new URL(response.url!), c);
+            });
+          }
+          return response;
+        },
+      ],
+    },
+  };
+
+  if (body) {
+    if (body instanceof Uint8Array) {
+      request.body = Buffer.from(body);
+    } else {
+      request.body = Buffer.from(JSON.stringify(body));
+    }
+  }
+
+  try {
+    const response = await got<Res>(request);
+    parseHeaders<Res>(response);
+    return parseResponse<Res>(response);
+  } catch (error) {
+    const gotError = error as RequestError;
+    if (gotError.response) {
+      return parseResponse<Res>(gotError.response as Response<Res>);
+    }
+    throw error;
+  }
+}
+
+function parseResponse<Res>(response: Response<Res>): AptosClientResponse<Res> {
+  return {
+    status: response.statusCode,
+    statusText: response.statusMessage || "",
+    data: response.body,
+    config: response.request.options,
+    request: response.request,
+    response,
+    headers: response.headers,
+  };
+}
+
+function parseHeaders<Res>(response: Response<Res>) {
+  // capitalize authorization key to be compatible with axios response
+  if (response.request.options.headers.authorization) {
+    response.request.options.headers.Authorization = response.request.options.headers.authorization;
+    delete response.request.options.headers.authorization;
+  }
+  // capitalize content-type key to be compatible with axios response
+  if (response.request.options.headers["content-type"]) {
+    response.request.options.headers["Content-Type"] = response.request.options.headers["content-type"];
+    delete response.request.options.headers["content-type"];
+  }
+}
+
+/**
+ * got supports only - string | number | boolean | null | undefined
+ * in the searchParam props,
+ * so if we have bigint type, convert it to Number
+ */
+function convertBigIntToNumber(obj: any): any {
```

Comment:
What happens if we need a value > the largest number?

Reply:
yea.. so largest Number == 9007199254740991, think it is big enough...?

Reply:
Is it used for the purposes of a counter or something just for management of the connection, or for information that's transferred on the wire associated with smart contract data?

Reply:
it is used in as a `query param` - so querying something like [ledger version](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/providers/aptos_client.ts#L140) or [event type creation number](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/providers/aptos_client.ts#L335) or in [pagination](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/providers/aptos_client.ts#L47) or [transaction version](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/providers/aptos_client.ts#L548) where `AnyNumber` can be a `number` or `BigInt` type

Reply:
or we use `string`
![image](https://github.com/aptos-labs/aptos-core/assets/29798064/7c98a850-f33a-4165-8f84-f10b61fd7435)

Reply:
Let's convert to string then.